network/book.go

```diff
@@ -177,42 +177,61 @@ func (s *SimpleBook) Failed(addr string) {
 	e.failure++
 }
 
-// Get will return the top priority peer out of all peers that are currently not active. It should
+// OrderedSample will return the top priority peer out of all peers that are currently not active. It should
 // represent the peer that is most likely to allow us to connect successfully.
-func (s *SimpleBook) Get() (string, error) {
+func (s *SimpleBook) OrderedSample(params ...int) ([]string, error) {
 	s.mutex.Lock()
 	defer s.mutex.Unlock()
 	entries := s.slice(false)
 	if len(entries) == 0 {
-		return "", errBookEmpty
+		return nil, errBookEmpty
 	}
 	sort.Sort(byPriority(entries))
-	e := entries[0]
-	e.active = true
-	return e.addr, nil
+
+	sampleSize := s.sampleSize
+	if len(params) > 0 {
+		sampleSize = params[0]
+	}
+	if len(entries) > sampleSize {
+	    entries = entries[:sampleSize]
+	}
+	
+	addrs := s.getAddresses(entries, sampleSize)
+	return addrs, nil
 }
 
-// Sample will return a random sample of peers that can be used to exchange peer addresses with
+// RandomSample will return a random sample of peers that can be used to exchange peer addresses with
 // other peers and allow discovery in the peer-to-peer network.
-func (s *SimpleBook) Sample() ([]string, error) {
+func (s *SimpleBook) RandomSample(params ...int) ([]string, error) {
 	s.mutex.Lock()
 	defer s.mutex.Unlock()
 	entries := s.slice(true)
+
+	sampleSize := s.sampleSize
+	if len(params) > 0 {
+		sampleSize = params[0]
+	}
+	
 	if len(entries) == 0 {
 		return nil, errors.New("no valid addresses")
 	}
-	if len(entries) > s.sampleSize {
+	if len(entries) > sampleSize {
```

Comment:
Yes, this means you have to do a cast to `int` for the comparison.
It's one thing I slightly dislike in Go; I think `len(...)` should return a `uint`.
However, casting like that for comparison with length is quite standard.